### ISSUE TITLE
DEMO: Resource (Module but maybe others) filtering is no longer working as it was before

### ISSUE THREAD

**Issue by AlexThurston** [usnistgov/ACVP#867]
**environment**
Demo

**testSessionId**
N/A

**vsId**
N/A

**Algorithm registration**
N/A

**Endpoint in which the error is experienced**
https://demo.acvts.nist.gov/acvp/v1/modules?limit=20&name=contains:Lightship&offset=0 GET

**Expected behaviour**
It seems as though the way filtering worked originally has changed.  The request above would normally work and does on production.  However, when done on demo, I get an HTTP status code for 416 (which does not seem like the correct HTTP response code, I would expect a 400) and a body of:
```json
[
  {
    "acvVersion": "1.0"
  },
  {
    "error": "Invalid filter applied: name=contains:Lightship"
  }
]
```

This appears to be the same error i get when the same route but with `vendors` instead.

The `limit` and `offset` query parameters appear to be working correctly.

**Comment by shaneshaffer** [comment]
Your querystring is missing the required [<index>] after the name. The spec says each query parameter element must be of the form <property>[<index>]=<operation>:<value>

https://demo.acvts.nist.gov/acvp/v1/modules?limit=20&name[0]=contains:Lightship&offset=0 should work correctly

**Comment by AlexThurston** [comment]
Perfect.  Just as a point of interest, this was not the case prior to the new API.  The query parameters do not currently require the `[<index>]` on production.  

Regardless, thanks for the clarification.

I also think that the HTTP status code 416 is a bit strange.  From the documentation of that code:

> A server SHOULD return a response with this status code if a request included a Range request-header field (section 14.35), and none of the range-specifier values in this field overlap the current extent of the selected resource, and the request did not include an If-Range request-header field. (For byte-ranges, this means that the first- byte-pos of all of the byte-range-spec values were greater than the current length of the selected resource.)
> 
> When this status code is returned for a byte-range request, the response SHOULD include a Content-Range entity-header field specifying the current length of the selected resource (see section 14.16). This response MUST NOT use the multipart/byteranges content- type.
> 

But in this case, this is a malformed or bad request (ie. 400), and not anything to do with the Range request-header.

**Comment by dmchodnicki** [comment]
Item is being Closed.  The root cause of the original problem was the missing `[<index>]` now consistently required in all environments since the new API has been deployed to Prod.  Improved error messaging is being tracked in other tickets.
